pennylane/templates/state_preparations.py

```diff
@@ -53,24 +58,36 @@ def BasisStatePreparation(basis_state, wires):
     r"""
     Prepares a basis state on the given wires using a sequence of Pauli X gates.
 
+    .. warning::
+
+        ``basis_state`` influences the circuit architecture and is therefore incompatible with
+        gradient computations. Ensure that ``basis_state`` is not passed to the qnode by positional
+        arguments.
+
     Args:
         basis_state (array): Input array of shape ``(N,)``, where N is the number of wires
             the state preparation acts on. ``N`` must be smaller or equal to the total
             number of wires of the device.
         wires (Sequence[int]): sequence of qubit indices that the template acts on
+
+    Raises:
+        QuantumFunctionError if inputs do not have the correct format.
     """
 
-    if not isinstance(wires, Iterable):
-        raise ValueError("Wires must be passed as a list of integers; got {}.".format(wires))
+    ######################
+    # Input checks
+    wires, n_wires = _check_wires(wires)
+    _check_shape(basis_state, (n_wires,))
 
-    if not len(basis_state) == len(wires):
-        raise ValueError(
-            "Number of qubits must be equal to the number of wires, which is {}; "
-            "got {}.".format(len(wires), len(basis_state))
-        )
+    # basis_state cannot be trainable
+    mssg = "Basis state influences circuit architecture and can therefore not be passed as a " \
+           "positional argument to the quantum node."
+    _check_no_variable([basis_state], ['basisstate'], mssg=mssg)
```

Comment:
Well, it contains 0 and 1 that decide whether or not there is an X flip or not. You cannot get a continuous gradient out of this. 

So better stop users from using this input as a variable, because then PennyLane will be able to start doing stupid things with it.